### PR TITLE
V0.1 transporters enhancement

### PR DESCRIPTION
**Pull request by AHReccese**
#### What does this implement/fix? Explain your changes.
* add deep recursive numpy.ndarray to list serializer in GeneralDataStructure transporter
* add deep recursive list to numpy.ndarray deserializer in GeneralDataStructure transporter
* update labelbinerizer transporter

### CODE REVIEW

**Comment by AHReccese** [comment]
I will add coverage-associated tests in the `NN support` PR because these new parts will be mainly used in NNs and fully covered in that PR.

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,10 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+- GeneralDataStructure transporter enhanced
+  - Deep ndarray to list serialization
+  - Deep list to ndarray deserialization 
+- LabelBinerizer transporter updated
```
Comment (sadrasabouri) [inline CHANGELOG.md:11]: Follow the pattern used in other projects (see [pycm's changelog](https://raw.githubusercontent.com/sepandhaghighi/pycm/master/CHANGELOG.md)). Something like below would be appropriate:
```
### Added
- `ndarray_to_list` method added to `GeneralDataStructureTransporter`
- `list_to_ndarray` method added to `GeneralDataStructureTransporter` 
### Changed
- GeneralDataStructure transporter enhanced
- LabelBinerizer transporter updated
```

**Review by sadrasabouri — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (AHReccese) [reply CHANGELOG.md:11]: Sure, applied.

**Comment by AHReccese** [comment]
> I will add coverage-associated tests in the `NN support` PR because these new parts will be mainly used in NNs and fully covered in that PR.

@sepandhaghighi 
are you ok with it?

**Review by sadrasabouri — APPROVED** [review]

**Comment by AHReccese** [comment]
> Great. The PR scope is well organized. Minor points needed to be mentioned:
> 
> * Update `CHANGELOG.md`
> * Add some test so that coverage percent increases (simple function test to make ensure us about the performance of these functions)
> * Added function's name is so long, make it shorter if it's possible
> * As you said in the discord channel, either solve the Python 3.5 problem or drop it. Anyway merge the PR to `dev` after passing all tests.

Also, Sepand announced his agreement to Python 3.5 drop.

**Pull request merged**